core/include/scipp/core/element/special_values.h

```diff
@@ -12,33 +12,43 @@
 
 namespace scipp::core::element {
 
+constexpr auto special_value_args = arg_list<int32_t, int64_t, double, float>;
+
+template <typename T>
+constexpr auto is_integer = std::numeric_limits<std::decay_t<T>>::is_integer;
```

Comment:
Not sure if it is good style to put things like `std::decay_t` into a trait? Maybe just use `std::is_integral` below, which is a bit shorter, and avoid defining the trait?

Or define a common base class:

```cpp
constexpr auto is_special_value =
    overloaded{arg_list<double, float, int64_t, int32_t>,
               [](const int64_t x) { return false; },
               [](const int32_t x) { return false; },
               [](const units::Unit &) { return units::dimensionless; }};
};
```
and use this as the based for `isnan` and friends?